salt/modules/swarm.py

```diff
@@ -30,10 +30,13 @@
 # Import Salt libs
 import salt.utils.json
 
+HAS_DOCKER = False
+
 try:
     import docker
 
-    HAS_DOCKER = True
+    if hasattr(docker, "from_env"):
```

Comment:
I would like to see this a standalone function for checking attributes that we use for this file.

The other approach that we could take is determining if this is a NamespacePackage vs an actual file. Though if we dump this check into a function, e.g. `HAS_DOCKER = _is_docker_module(docker)`, we could do multiple checks to verify that the docker module looks like what it should look like.

The advantage of that is easily being able to test the function in question, and being able to expand or narrow the definition as needed.

Reply:
Thanks for the suggestions @waynew - I've introduce that function and added unit tests :+1: